examples/drm.rs

```diff
@@ -0,0 +1,186 @@
+            .crtcs()
+            .iter()
+            .map(|&crtc| device.get_crtc(crtc))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        // Find a connected crtc.
+        let con = connectors
+            .iter()
+            .find(|con| con.state() == connector::State::Connected)
+            .ok_or("No connected connectors")?;
+
+        // Get the first CRTC.
+        let crtc = crtcs.first().ok_or("No CRTCs")?;
+
+        // Find a mode to use.
+        let mode = con
+            .modes()
+            .iter()
+            .find(|mode| mode.mode_type().contains(ModeTypeFlags::PREFERRED))
+            .or_else(|| con.modes().first())
+            .ok_or("No modes")?;
+
+        // Look for a primary plane compatible with our CRTC.
+        let planes = device.plane_handles()?;
+        let planes = planes
+            .iter()
+            .filter(|&&plane| {
+                device.get_plane(plane).map_or(false, |plane| {
+                    let crtcs = handles.filter_crtcs(plane.possible_crtcs());
+                    crtcs.contains(&crtc.handle())
+                })
+            })
+            .collect::<Vec<_>>();
+
+        // Find the first primary plane or take the first one period.
+        let plane = planes
+            .iter()
+            .find(|&&&plane| {
+                if let Ok(props) = device.get_properties(plane) {
+                    let (ids, vals) = props.as_props_and_values();
+                    for (&id, &val) in ids.iter().zip(vals.iter()) {
+                        if let Ok(info) = device.get_property(id) {
+                            if info.name().to_str().map_or(false, |x| x == "type") {
+                                return val == PlaneType::Primary as u32 as u64;
+                            }
+                        }
+                    }
+                }
+
+                false
+            })
+            .or(planes.first())
+            .ok_or("No planes")?;
+
+        // Create the surface on top of this plane.
+        // Note: This requires root on DRM/KMS.
+        let mut surface = unsafe {
+            Surface::from_raw(&context, {
+                let mut handle = DrmWindowHandle::empty();
+                handle.plane = (**plane).into();
+                handle.into()
+            })
+        }?;
+
+        // Resize the surface.
+        let (width, height) = mode.size();
+        surface.resize(
+            NonZeroU32::new(width as u32).unwrap(),
+            NonZeroU32::new(height as u32).unwrap(),
+        )?;
+
+        // Start drawing to it.
+        let start = Instant::now();
+        let mut tick = 0;
+        while Instant::now().duration_since(start) < Duration::from_secs(2) {
+            tick += 1;
+            println!("Drawing tick {tick}");
+
+            // Start drawing.
+            let mut buffer = surface.buffer_mut()?;
+            draw_to_buffer(&mut buffer, tick);
+            buffer.present()?;
+
+            // Wait for the page flip to happen.
```

Comment:
I don't think this actually waits on anything currently. To "wait for the page flip" it would have to `poll` the fd and read any events when it's readable until `Event::PageFlip` occurs, right?